modules/core/src/main/java/org/apache/ignite/internal/lang/Pair.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.lang;
+
+import java.util.Objects;
+
+/**
+ * Pair of objects.
+ *
+ * @param <T> First object.
+ * @param <V> Second object.
+ */
+public class Pair<T, V> {
```

Comment:
This the 4th Pair class in our codebase, while we have IgniteBiTuple

Reply:
follow-up https://issues.apache.org/jira/browse/IGNITE-24492

Reply:
Would it be possible not to use pairs and tuples at all? It is currently very difficult to read the code containing something like `IgniteBiTuple<UUID, Boolean>`, what's wrong with introducing domain-specific classes with two fields?